### PR TITLE
use regexp to parse Linux VM names

### PR DESCRIPTION
**Pull request by dmitsh**
**What this PR does / why we need it**: use regexp to parse Linux VM names

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/armhelpers/util.go [hunk 1]
```diff
@@ -3,13 +3,34 @@ package armhelpers
...
 )
 
+const (
+	// TODO: merge with the RP code
+	// here there are 4 capture groups
+	//  #1 is the orchestrator (dcos|k8s|swarm|swarmm)
+	//  #2 is the agent pool name, which can contain -s
+	//  #3 is the Cluster ID for the cluster
+	//  #4 is the index of the vm in the agent pool/master pool
+	k8sLinuxVMNamingFormat         = "^([^-]+)-(.+)-([0-9a-fA-F]{8})-{0,2}([0-9]+)$"
```
Comment (amanohar) [inline pkg/armhelpers/util.go]: If it's k8s we don't need to generalize the first part IMO. We already have an assumption in upgradecluster.go and all the BI.

File: pkg/armhelpers/util.go [hunk 2]
```diff
@@ -39,19 +60,21 @@ func SplitBlobURI(URI string) (string, string, string, error) {
 
 // LinuxVMNameParts returns parts of Linux VM name e.g: k8s-agentpool1-11290731-0
 func LinuxVMNameParts(vmName string) (orchestrator string, poolIdentifier string, nameSuffix string, agentIndex int, err error) {
-	vmNameParts := strings.Split(vmName, "-")
-
-	if len(vmNameParts) != 4 {
+	vmNameParts := vmnameLinuxRegexp.FindStringSubmatch(vmName)
+	if len(vmNameParts) != 5 {
```
Comment (amanohar) [inline pkg/armhelpers/util.go]: This needs to remove. The point is it can be more than five.

Comment (dmitsh) [reply pkg/armhelpers/util.go]: no, it cannot

File: pkg/armhelpers/util.go (hunk 1 above)
Comment (dmitsh) [reply pkg/armhelpers/util.go]: plz explain

File: pkg/armhelpers/util.go (hunk 2 above)
Comment (amanohar) [reply pkg/armhelpers/util.go]: Discussed offline. This can remain.

File: pkg/armhelpers/util.go (hunk 1 above)
Comment (amanohar) [reply pkg/armhelpers/util.go]: Discussed offline

**Pull request merged**